packages/manager/modules/emailpro/src/service/messaging/emailpro.messaging.service.js

```diff
@@ -0,0 +1,177 @@
+import { CSS_BOOTSTRAP_CLASSES } from './emailpro.messaging.constants';
+
+const isEmptyObj = (passedObj) =>
+  !(
+    passedObj &&
+    passedObj === Object(passedObj) &&
+    Object.keys(passedObj).length !== 0
+  );
+const uniqBy = (arr, fn, set = new Set()) =>
+  arr.filter((el) =>
+    ((v) => !set.has(v) && set.add(v))(
+      typeof fn === 'function' ? fn(el) : el[fn],
+    ),
+  );
+
+export default class Messaging {
+  resetMessages() {
+    this.message = null;
+    this.messageDetails = null;
+  }
+
+  write(message, alertType, failure = null) {
+    this.message = message;
+    this.alertType = `alert ${alertType}`;
+    this.messageDetails = [];
+
+    if (failure != null) {
+      const value = [failure, failure.data];
+
+      value.forEach((currentValue) => {
+        if (!isEmptyObj(currentValue)) {
+          if (typeof currentValue === 'string') {
+            this.messageDetails.push({ id: null, message: currentValue });
+          } else if (typeof currentValue.message === 'string') {
+            this.messageDetails.push({
+              id: currentValue.id,
+              message: currentValue.message,
+            });
+          } else if (currentValue.messages != null) {
+            this.messageDetails = currentValue.messages.map(
+              (currentMessage) => ({
+                id: currentMessage.code,
+                message: currentMessage.message,
+              }),
+            );
+            this.messageDetails = uniqBy(
+              this.messageDetails,
+              (currentMessage) =>
+                `${currentMessage.id}${currentMessage.message}`,
+            );
```

Comment:
```suggestion
            this.messageDetails = this.messageDetails.reduce(
                (uniqueMessages, item) => { 
                   if (!uniqueMessages.find(uniqueMessage => 
                     item.id === uniqueMessage.id && item.message === uniqueMessage.message
                   )) {
                     uniqueMessages.push(item);
                   }
                return uniqueMessages;
            }, []);
```
Maybe you can remove the uniqBy function and use this instead as it is used only once here ?